server/src/api/customers.ts

```diff
@@ -83,4 +84,38 @@ customerRouter.post(
   }
 );
 
+/**
+ * Handles the PATCH requests to modify an existing customer with the specified customerId.
+ * If successful, returns the modified customer in the response body with code 200.
+ */
+customerRouter.patch(
+  '/:customerId',
+  customerAuth,
+  validateAndFormatPhoneNumber(
+    (body: Partial<CustomerPayload>) => body.contactNumber,
+    (body: Partial<CustomerPayload>, phoneNumber: string) =>
+      (body.contactNumber = phoneNumber)
+  ),
+  async (req: Request, res: Response, next: NextFunction) => {
+    const {customerId: customerIdStr} = req.params;
+    const customerId = Number(customerIdStr);
+    const customerData = req.body;
```

Comment:
Do you think `customerData` may sound like it contains all the fields of a Customer and not just those to be updated?

Reply:
@karenfrilya97 Okay will change that!